functions/Get-DbaJobOutputFile.ps1

```diff
@@ -0,0 +1,201 @@
+param
+(# The Server/instance 
+        [Parameter(Mandatory=$true,HelpMessage='The SQL Server Instance', 
+                   ValueFromPipeline=$true,
+                   ValueFromPipelineByPropertyName=$true, 
+                   ValueFromRemainingArguments=$false, 
+                   Position=0)]
+        [ValidateNotNull()]
+        [ValidateNotNullOrEmpty()]
+        [object]$SqlServer,
+        [Parameter(Mandatory=$false,HelpMessage='SQL Credential', 
+                   ValueFromPipeline=$true,
+                   ValueFromPipelineByPropertyName=$true, 
+                   ValueFromRemainingArguments=$false, 
+                   Position=1)]
+        [System.Management.Automation.PSCredential]$SqlCredential,
+        [Parameter(Mandatory=$false,HelpMessage='Want to open the file')] 
+        [switch]$OpenFile)
+    DynamicParam { if ($SqlServer) { return (Get-ParamSqlJobs -SqlServer $SqlServer -SqlCredential $SourceSqlCredential) } }
+
+	BEGIN 
+    {
+		$jobname = $psboundparameters.Jobs
+        # Create Table Object
+         $table = New-Object system.Data.DataTable $TableName 
+         # Create Columns
+         $col0 = New-Object system.Data.DataColumn Server,([string])
+         $col1 = New-Object system.Data.DataColumn Instance,([string])
+         $col2 = New-Object system.Data.DataColumn Job,([string])
+         $col3 = New-Object system.Data.DataColumn JobStep,([string])
+         $col4 = New-Object system.Data.DataColumn FileName,([string])
+
+         #Add the Columns to the table
+         $table.columns.add($col0)
+         $table.columns.add($col1)
+         $table.columns.add($col2)
+         $table.columns.add($col3)
+         $table.columns.add($col4)
+        function Add-Row 
+        {
+        # Create a new Row
+            $row = $table.NewRow() 
+            # Add values to new row
+            $row.Server = $Server.ComputerNamePhysicalNetBIOS
+            $row.Instance = $Server.Instancename
+            $row.Job = $Job.Name
+            $row.JobStep = $step.Name
+            $row.FileName = $FileName
+          
+            #Add new row to table
+            $table.Rows.Add($row)
+        }
+         
+    }
+    PROCESS
+    {
+        foreach ($instance in $sqlserver)
+        {
+            $Server = Connect-SqlServer -SqlServer $instance  -SqlCredential $SqlCredential
+            $jobs = $Server.JobServer.Jobs
+            if ($JobName)
+            {
+                $Job = $server.JobServer.Jobs[$JobName]
+                $Servername = $Server.ComputerNamePhysicalNetBIOS
+                foreach($Step in $Job.JobSteps)
+                {
+                $fileName = $Step.OutputFileName
+                if($fileName -eq '')
+                {
+                    Add-Row 
+                }
+                else
+                {         
+                    if($FileName.StartsWith('\\') -eq $false)
+            {
+                $fileName = '\\' + $Servername + '\' + $Filename.Replace(':','$')
```

Comment:
We have Join-AdminUnc internal command to help with this :) Please use instead! Join-AdminUnc $servername $filename I think